node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeReports.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.configserver.noderepository;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.TreeMap;
+
+import static com.yahoo.yolean.Exceptions.uncheck;
+
+/**
+ * API of node reports within node-admin.
+ *
+ * @author hakonhall
+ */
+public class NodeReports {
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final Map<String, JsonNode> reports = new TreeMap<>();
+
+    public NodeReports() { }
+
+    public NodeReports(NodeReports reports) {
+        this.reports.putAll(reports.reports);
+    }
+
+    private NodeReports(Map<String, JsonNode> reports) {
+        this.reports.putAll(reports);
+    }
+
+    public static NodeReports fromMap(Optional<Map<String, JsonNode>> reports) {
```

Comment:
This `Optional` check should be done by the caller, most of the users make a redundant wrap in `Optional`...

Reply:
Fixed